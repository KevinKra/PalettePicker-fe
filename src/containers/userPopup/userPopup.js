import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import * as actions from "../../_redux/actions";
import { loginUser, createUser } from "../../utilities/apiCalls";
import "./userPopup.scss";

class userPopup extends Component {
  state = {
    username: "",
    password: ""
  };

  handleExit = () => {
    this.props.history.push("/");
  };

  handlePageChange = destination => this.props.history.push(`/${destination}`);

  // user logs in, get user data / projects from server
  // dispatch data to store
  // close popup window
  handleLogin = async e => {
    console.log("logging in...");
    const { username, password } = this.state;
    e.preventDefault();
    try {
      const user = await loginUser(username, password);
      this.props.updateCurrentUser(user);
      this.handleExit();
    } catch (error) {
      console.log(error.message);
      this.clearInputs();
    }
  };

  handleRegister = e => {
    console.log("Registering...");
    e.preventDefault();
    // capture register values
    // make sure user doesnt already exist in DB
    // Good: add user to database
    // REDUX update user => ACTION: this.props.updateCurrentUser(userDetails)
    // login user / close window
  };

  handleChanges = e => {
    const { name, value } = e.target;
    this.setState({ [name]: value });
  };

  clearInputs = () => {
    this.setState({ username: "", password: "" });
  };

  render() {
    const inputs = (
      <Fragment>
        <label htmlFor="username">Enter Username</label>
        <input
          type="text"
          name="username"
          value={this.state.username}
          onChange={e => this.handleChanges(e)}
        />
        <label htmlFor="password">Enter Password</label>
        <input
          type="password"
          name="password"
          value={this.state.password}
          onChange={e => this.handleChanges(e)}
        />
      </Fragment>
    );
    const loginFormat = (
      <Fragment>
        <h3 className="editor-title">To continue, log in to PalettePicker.</h3>
        <form className="user-form" onSubmit={e => this.handleLogin(e)}>
          {inputs}
          <div className="form-btns">
            <button className="form-btn">Login</button>
            <button
              className="form-btn"
              onClick={() => this.handlePageChange("register")}
            >
              new user?
            </button>
          </div>
        </form>
      </Fragment>
    );

    const registerFormat = (
      <Fragment>
        <h3 className="editor-title">Sign up with a username and password.</h3>
        <form className="user-form" onSubmit={e => this.handleRegister(e)}>
          {inputs}
          <div className="form-btns">
            <button className="form-btn">Register</button>
            <button
              className="form-btn"
              onClick={() => this.handlePageChange("login")}
            >
              Already have an account?
            </button>
          </div>
        </form>
      </Fragment>
    );
    return (
      <Fragment>
        <div className="screen" />
        <section className="UserPopup">
          <button className="popup-exit" onClick={this.handleExit}>
            X
          </button>
          <section className="form-section">
            {this.props.location.pathname === "/login"
              ? loginFormat
              : registerFormat}
          </section>
        </section>
      </Fragment>
    );
  }
}

const mapDispatchToProps = dispatch => ({
  addUserProjects: projects => dispatch(actions.addUserProjects(projects)),
  updateCurrentUser: userDetails =>
    dispatch(actions.updateCurrentUser(userDetails))
});

export default connect(
  null,
  mapDispatchToProps
)(userPopup);
