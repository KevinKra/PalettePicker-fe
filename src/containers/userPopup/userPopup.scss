.UserPopup {
  position: fixed;
  margin-left: auto;
  margin-right: auto;
  top: 25%;
  left: 15%;
  right: 15%;
  z-index: 4;
  max-width: 400px;
  border: 1px solid coral;
  color: red;

  .palettes-section {
    border: 1px solid red;
    height: 300px;
    width: 80%;
    overflow: scroll;
  }
  .palette-details {
    display: flex;
    justify-content: space-between;
    margin: 0.2rem 0.4rem;
    border: 1px solid blue;
  }
  .palette-inputs {
    display: flex;
  }
  .palette-color {
    height: 1.5rem;
    min-width: 5rem;
    margin-right: 0.5rem;
    border: 1px solid black;
  }
  .editor-form {
    display: flex;
    flex-direction: column;
    align-items: center;
    background-color: white;
  }

  .editor-title {
    margin: 0;
  }

  .popup-exit {
    position: absolute;
    right: -10px;
    top: -10px;
  }

  .submit-btn {
    border: 1px solid red;
    background-color: coral;
    color: white;
    font-size: 16px;
    border-radius: 5px;
    width: 70%;
    padding: 0.5rem;
    margin: 0.5rem;
  }

  .remove-btn {
  }
}
.screen {
  position: absolute;
  top: 0;
  z-index: 3;
  background-color: rgba(6, 16, 20, 0.658);
  height: 100%;
  width: 100vw;
}
