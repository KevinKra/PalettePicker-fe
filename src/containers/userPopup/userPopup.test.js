import React from "react";
import { shallow } from "enzyme";
import { userPopup } from "./userPopup";

describe("userPopup", () => {
  it("matches snapshot", () => {
    // const props = {
    //   currentPalette: [
    //     {
    //       locked: true
    //     }
    //   ],
    //   userProjects: [{ projectId: 8 }]
    // };
    const wrapper = shallow(<userPopup />, {
      disableLifecycleMethods: true
    });
    expect(wrapper).toMatchSnapshot();
  });
});
