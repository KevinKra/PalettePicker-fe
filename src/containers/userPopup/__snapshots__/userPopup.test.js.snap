// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`userPopup matches snapshot 1`] = `<userPopup />`;
